### DISCUSSION TITLE
u8x8 font modification on the fly, adding fx

### DISCUSSION THREAD

**Discussion by ilovesushi** [olikraus/u8g2#2512] [Q&A] [answered]
first question
using u8x8 due to small memory foot print. need to add underscore upperscores, draw frames around text or text labels , and probably draw fx on top. how can i read a symbol from a (current)font in a tile buffer for futher modding and printing.
second question
were unable to find any relative info how to load 4x6 font into u8x8 and display, anyway i am planing to construct a tile 8x8 from 4x6 font to compress the output, so how to load font and get the tile buffer of it
appreciate

**Comment by olikraus** [comment]
first question: 

There is a function
``` C++
void u8x8_get_glyph_data(u8x8_t *u8x8, uint8_t encoding, uint8_t *buf, uint8_t tile_offset) 
```
which will copy the 8x8 tile into the provided buffer. In Arduino, it might look like this:
```
uint8_t buf[8];
u8x8_get_glyph_data(u8x8.getU8x8(), 'A', buf, 0);
/* modify the tile in buf */
u8x8.drawTile(1, 2, 1 buf);
```
There is also an Arduino example, which demonstrates how to rotate a 8x8 tile by 90 degree: 
https://github.com/olikraus/u8g2/blob/416abe4af4e5afd10045c42e46c0d844b6761ddc/sys/arduino/u8x8/Rotate90/Rotate90.ino#L362-L367

The Rotate90.ino example is also available in Arduino IDE of the latest u8g2 release.

second question:
There is no 4x6 font. You need to convert this into 8x8 tiles.

**Reply by ilovesushi** [reply]
You are the best. Thanks. 
Regarding the second question. I scrolled the font page for u8g2 , there are a lot of 4x6,5x7 fonts, seems I am missing something. They can't be loaded directly to u8x8 because they are for u8g2, but if I could I would draw them as tiles so I would be able to draw 3 or 4 glyph in 8,or 16 space

**Reply by olikraus** [reply, marked as answer]
The font format differs between u8g2 and u8x8, but you can convert a 4x6 font into u8x8 format. However Each char would be 8 pixel wide. 

As an alternative you could use u8g2 in page mode. It will only consume a little bit more RAM. Even better, u8g2 fonts are compressed.

**Reply by ilovesushi** [reply]
how can i convert a 4x6 font into u8x8 format. and load with u8x8 separately. i am perfectly fine with using gliph and tile functions only to cut out smaller letters

**Reply by olikraus** [reply]
There is some help in the FAQ: https://github.com/olikraus/u8g2/blob/eba8b1a0190ef4d9b3b3fd0ec907d9c039b31e9e/doc/faq.txt#L255-L289
To produce u8x8 fonts you need to use "-f 2" instead.
The bdf source is here: https://github.com/olikraus/u8g2/blob/master/tools/font/bdf/4x6.bdf
